pkg/extension/sumologicextension/extension.go

```diff
@@ -148,43 +148,82 @@ func (se *SumologicExtension) validateCredenials(
 
 func (se *SumologicExtension) Start(ctx context.Context, host component.Host) error {
 	se.logger.Info(banner)
-	colCreds, registrationDone, err := se.getCredentials(ctx)
+
+	colCreds, registrationDone, err := se._r(ctx, host)
 	if err != nil {
 		return err
 	}
 
+	if !registrationDone {
+		se.logger.Info("Checking if locally retrieved credentials are still valid...",
+			zap.String(collectorNameField, colCreds.Credentials.CollectorName),
+			zap.String(collectorIdField, colCreds.Credentials.CollectorId),
+		)
+
+		if err := se.validateCredenials(ctx, colCreds.Credentials); err != nil {
+			se.logger.Info("Locally stored credentials invalid. Trying to re-register...")
+			if err = se.credentialsStore.Delete(se.hashKey); err != nil {
+				return err
+			}
+
+			// Credentials might have ended up being invalid or the collector
+			// might have been removed in Sumo.
+			// Fall back to removing the credentials and recreating them.
+			colCreds, _, err = se._r(ctx, host)
+			if err != nil {
+				return fmt.Errorf("registration failed: %w", err)
+			}
+			se.logger.Info("Registration finished successfully",
+				zap.String(collectorNameField, colCreds.Credentials.CollectorName),
+				zap.String(collectorIdField, colCreds.Credentials.CollectorId),
+			)
+		} else {
+			se.logger.Info("Local collector credentials all good, starting up the collector",
+				zap.String(collectorNameField, colCreds.Credentials.CollectorName),
+				zap.String(collectorIdField, colCreds.Credentials.CollectorId),
+			)
+		}
+	}
+
 	// Add logger fields based on actual collector name and ID as returned
 	// by registration API.
 	se.logger = se.logger.With(
 		zap.String(collectorNameField, colCreds.Credentials.CollectorName),
 		zap.String(collectorIdField, colCreds.Credentials.CollectorId),
 	)
 
+	go se.heartbeatLoop()
+
+	return nil
+}
+
+func (se *SumologicExtension) _r(ctx context.Context, host component.Host) (credentials.CollectorCredentials, bool, error) {
```

Comment:
what does mean `_r`?

Reply:
I guess `register`, but would change name to more informative

Reply:
Forgot to change it 😅 . PTAL now